pkg/api/handlers.go

```diff
@@ -111,7 +111,17 @@ func indexHandler(httpPathPrefix string, content *IndexPageContent) http.Handler
 	}
 }
 
-func configHandler(actualCfg interface{}, defaultCfg interface{}) http.HandlerFunc {
+func (cfg *Config) configHandler(actualCfg interface{}, defaultCfg interface{}) http.HandlerFunc {
```

Comment:
I'm curious (again) why this needs to be lazy/late evaluated (Not a leading question, genuinely curious) - might be nice not to have to do this unless really necessary, as there becomes a burden to support dynamically swapping out the provider.

I can see that downstream implementations would have to be careful to not e.g. start serving requests with one implementation, and then change the implementation, which would lead to confusion for the user having different results.

Reply:
> I'm curious (again) why this needs to be lazy/late evaluated (Not a leading question, genuinely curious) - might be nice not to have to do this unless really necessary, as there becomes a burden to support dynamically swapping out the provider.

Right, I think it's not necessary to evaluate the handler lazily, as long as the module in the downstream project that override the handler has a dependency on the API module.

> I can see that downstream implementations would have to be careful to not e.g. start serving requests with one implementation, and then change the implementation, which would lead to confusion for the user having different results.

That's definitely not what one want! :D